### PR TITLE
Draft PR : Added ExpoSQLiteAdapter and Code Sharing for common files

### PR DESCRIPTION
**Pull request by chintannp**
#### Checklist

- [x] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by cshfang — CHANGES_REQUESTED** [review]
Mostly lgtm other than some nits but I think the `batchSave` might have a bug still. Please take a look

File: packages/datastore-storage-adapter/package.json [hunk 1]
```diff
@@ -34,9 +34,12 @@
...
 	"devDependencies": {
-		"@aws-amplify/core": "4.4.0",
-		"@aws-amplify/datastore": "3.7.7",
-		"react-native-sqlite-storage": "5.0.0"
+		"@aws-amplify/core": "4.3.14",
+		"@aws-amplify/datastore": "3.7.6",
+		"@types/react-native-sqlite-storage" : "5.0.1",
+		"expo-sqlite": "10.1.0",
+		"react-native-sqlite-storage": "5.0.0",
+		"expo-file-system" : "13.1.4"
```
[the remark this answers is not in the document]
Comment (chintannp) [reply packages/datastore-storage-adapter/package.json:42]: Thanks. Will update that.

File: packages/datastore-storage-adapter/webpack.config.js [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
 		'aws-amplify-datastore-storage-adapter.min': './lib-esm/index.js',
+		'aws-amplify-datastore-storage-adapter-expo.min':
+			'./lib-esm/ExpoSQLiteAdapter/ExpoSQLiteAdapter.js',
 	},
 	externals: [
 		'@aws-amplify/datastore',
 		'@aws-amplify/core',
 		'react-native-sqlite-storage',
+		'expo-sqlite',
+		'expo-file-system',
```
[the remark this answers is not in the document]
Comment (chintannp) [reply packages/datastore-storage-adapter/webpack.config.js]: Thanks. I have updated it.

File: packages/datastore-storage-adapter/src/ExpoSQLiteAdapter/ExpoSQLiteDatabase.ts [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+												logger.warn(error);
+												return true;
+											}
+										)
+									)
+							)
+						);
+					});
+				}
+				resolveTransaction(null);
```
[the remark this answers is not in the document]
Comment (chintannp) [reply packages/datastore-storage-adapter/src/ExpoSQLiteAdapter/ExpoSQLiteDatabase.ts]: I have updated `batchSave` to use single transaction.

**Pull request closed without merging**